### PR TITLE
bochs: 2.6.8 -> 2.6.9

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

Update

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/virtualization/bochs/default.nix [hunk 1]
```diff
@@ -1,10 +1,49 @@
-{ stdenv, fetchurl, config
+{ stdenv, fetchurl
 , pkgconfig, libtool
 , gtk2, mesa, readline, libX11, libXpm
 , docbook_xml_dtd_45, docbook_xsl
 , sdlSupport ? true, SDL2 ? null
-, termSupport ? true , ncurses ? null
+, termSupport ? true, ncurses ? null
 , wxSupport ? true, wxGTK ? null
+# A plethora of configurable options
```
Comment (FRidh) [inline pkgs/applications/virtualization/bochs/default.nix]: Let's not do this, its already possible to just override `configureFlags` with `overrideAttrs`.

Comment (AndersonTorres) [inline pkgs/applications/virtualization/bochs/default.nix]: So I will write a giant configureFlags!

File: pkgs/applications/virtualization/bochs/default.nix [hunk 2]
```diff
@@ -100,48 +57,50 @@ stdenv.mkDerivation rec {
...
+         "--enable-a20-pin"
+         "--enable-x86-64"
+         "--enable-smp"
+         "--enable-large-ramfile"
+         "--enable-repeat-speedups"
+         "--enable-handlers-chaining"
+         "--enable-trace-linking"
+         "--enable-configurable-msrs"
+         "--enable-show-ips"
+         "--enable-debugger"
```
Comment (Mic92) [inline pkgs/applications/virtualization/bochs/default.nix]: I get:

```
checking for repeated IO and mem copy speedups... yes
checking for gcc / MSVC nmake fast function calls optimization... no
checking for handlers chaining speedups... yes
checking for trace linking speedups support... yes
checking support for configurable MSR registers... yes
checking show IPS... yes
checking for use of .cpp as suffix... no
checking for Bochs internal debugger support... yes
checking for flex... not_found
checking for yacc... not_found
checking whether to generate parser / lexer files... no
checking for disassembler support... yes
checking enable Bochs internal debugger GUI... yes
checking for gdb stub enable... yes
configure: error: --enable-debugger and --enable-gdb-stub are mutually exclusive
builder for ‘/nix/store/155nqasd6hwck25maxdwl6dikc2i1cq9-bochs-2.6.9.drv’ failed with exit code 1
error: build of ‘/nix/store/155nqasd6hwck25maxdwl6dikc2i1cq9-bochs-2.6.9.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
The invocation of "nix-shell -p bochs" failed
$ git worktree prune
python3 review.py 27278  46.89s user 4.41s system 51% cpu 1:39.71 total
```

Comment (AndersonTorres) [inline pkgs/applications/virtualization/bochs/default.nix]: Oh, frak! The various options are conflicting. I will fix this.

**Comment by Mic92** [comment]
Is this error a critical one?

```
$ bochs
========================================================================                                                                                                                      
                       Bochs x86 Emulator 2.6.9                                                                                                                                               
               Built from SVN snapshot on April 9, 2017                                                                                                                                       
                  Compiled on Apr  9 2017 at 06:46:13                                                                                                                                         
========================================================================                                                                                                                      
00000000000i[      ] LTDL_LIBRARY_PATH not set. using compile time default '/nix/store/l6m5mzz4w099wbc35bndpa9qcrxpcxm5-bochs-2.6.9/lib/bochs/plugins'                                        
00000000000i[      ] BXSHARE not set. using compile time default '/nix/store/l6m5mzz4w099wbc35bndpa9qcrxpcxm5-bochs-2.6.9/share/bochs'                                                        
00000000000i[      ] lt_dlhandle is (nil)                                                                                                                                                     
00000000000p[      ] >>PANIC<< dlopen failed for module 'usb_common' (libbx_usb_common.so): file not found                                                                                    
00000000000e[SIM   ] notify called, but no bxevent_callback function is registered                                                                                                            
00000000000e[SIM   ] notify called, but no bxevent_callback function is registered                                                                                                            
========================================================================                                                                                                                      
Bochs is exiting with the following message:                                                                                                                                                  
[      ] dlopen failed for module 'usb_common' (libbx_usb_common.so): file not found                                                                                                          
========================================================================                                                                                                                      
00000000000i[SIM   ] quit_sim called with exit code 1      
```

**Comment by AndersonTorres** [comment]
Yes, it is; Bochs support for plugins is buggy, I disabled it.
(Indeed, the compile-time messages comply about dlopen and libtool...)

**Comment by AndersonTorres** [comment]
A question: can I add a "dlxlinux demo" package?

Normally, the precompiled Windows package for Bochs ships a small testing image with a simple Linux distro inside (dlxlinux), to use as demo/test. 

I think it would be cool to include a `dlxlinux.nix` package for it as a proof-of-concept. But maybe it is superfluous and completely optional.

What do you think?

**Comment by Mic92** [comment]
Mhm. Shouldn't it be trivial to download the image on its own, also the images seems to be pretty out-dated: https://sourceforge.net/projects/bochs/files/Disk%20Images/

**Comment by AndersonTorres** [comment]
Yes, the image is trivial indeed.

**Pull request merged**